pykern/pkcollections.py

```diff
@@ -99,9 +101,68 @@ def pkdel(self, name, default=None):
             except KeyError:
                 pass
 
+    def pkmerge(self, to_merge, make_copy=True):
+        """Add `to_merge` to `self`
+
+        Types are assumed to match and are not converted, e.g. dict is
+        not converted to PKDict. Again, use `canonicalize` if you want
+        to avoid type incompatibilities.
+
+        `to_merge`'s values override `self`'s so if say, `to_merge` is ``{'x': None}``,
+        then ``self.x`` will be `None` at the end of this call even if it had a value
+        before.
+
+        Lists from to_merge are prepended on this same principle, that
+        is, to_merge "overrides" self, and prepending is defined as
+        overriding. Lists must contain unique elements and duplicates will
+        cause an error.
+
+        This function recurses only on PKDicts.
+
+        Args:
+            to_merge (dict): elements will be copied into `self`
+            make_copy (bool): deepcopy `to_merge` before merging [True]
+
+        Returns:
+            PKDict: self
+        """
+
+        def _type_err(key, base, merge):
+            return AssertionError(
+                f"key={key} type mismatch between (self) base={base} and to_merge={merge}"
+            )
+
+        if make_copy:
+            to_merge = copy.deepcopy(to_merge)
+
+        for k in list(to_merge.keys()):
+            t = to_merge[k]
```

Comment:
is `t` necessary?

Reply:
I think so, because it is used 10+ times, and `to_merge[k]` is a function call (under the covers) so this type of caching is important in recursive routine.